src/cert_context.rs

```diff
@@ -107,6 +107,47 @@ impl CertContext {
         self.get_encoded_bytes()
     }
 
+    /// Deletes the key container for both CAPI and CNG keys
+    pub fn del_key_container(&mut self) -> io::Result<()>{
+        unsafe {
+            let bytes = self.get_bytes(Cryptography::CERT_KEY_PROV_INFO_PROP_ID)?;
+            assert!(bytes.len() <= u32::max_value() as usize);
+
+            let key_prov = bytes.as_ptr() as *const Cryptography::CRYPT_KEY_PROV_INFO;
+            
+            if (*key_prov).dwKeySpec == 0 {
+                // CNG Key
+
+                let mut prov_handle = Cryptography::NCRYPT_PROV_HANDLE::default();
+                let mut key_handle = Cryptography::HCRYPTPROV_OR_NCRYPT_KEY_HANDLE::default();
+                
+                Cryptography::NCryptOpenStorageProvider(&mut prov_handle, (*key_prov).pwszProvName, 0);
```

Comment:
For these APIs:
let mut ret = Cryptography::NCrypt()
if ret != 0 {
    return Err(io::Error::from_raw_os_error(ret));
 }

Reply:
Added